nntrainer/include/neuralnet.h

```diff
@@ -71,6 +71,11 @@ class NeuralNetwork {
   friend class ModelLoader; /** access private members of ModelLoader */
 
 public:
+  using NodeType = std::shared_ptr<Layer>; /** Type of a Node */
+  using GraphType = std::vector<NodeType>; /** actual graph type */
+  using FlatGraphType =
+    std::vector<NodeType>; /** topological sorted, iterable 1-D list of nodes */
```

Comment:
I'm wonder that if we are going to have other graph type.?? Is there any others and if it is, let me know what for?

Reply:
FlatGraphType is just a mere view (or reference) of a graph to iterate the graph in 1-D fashion. It is needed to make a multiple epoch test for now. Later it can work as a special view to access the graph easier.

Reply:
@jijoongmoon Later, I think this can be updated to an iterator which will iterate over the graph in the defined order by the user.